scanner/decisions/0010-separate-versioning-for-vuln-bundles.md

```diff
@@ -0,0 +1,118 @@
+
+Each bundle is created with an Exporter running the same ClairCore version used to build the release, ensuring compatibility.  The Exporter serves bundles at `https://definitions.stackrox.io/v4/<version>/vulnerabilities.zip` where `<version>` is the supported StackRox release (such as "4.5.0").  Scanner uses its release version tagged during build time to determine which URL to fetch from.
+
+In offline mode, one offline bundle per release was introduced for Scanner V4.  Coexisting with the "latest" Scanner V2 offline bundles.  To facilitate user migration to per-release offline bundles, the latest offline bundle used by Scanner V2 (at the time of this writing served at https://install.stackrox.io/scanner/scanner-vuln-updates.zip) was changed to contain Scanner V4's offline bundles for `4.[45][.x]` releases.  A command `roxctl scanner download-db` was created to facilitate users fetching the right offline bundle for their release.
+
+This approach needs to be revised for the following reasons:
+
+1.  The vulnerability schema rarely changes, but maintaining multiple bundles is expensive.  It requires multiple GitHub workflow runs, which increases the chances of failure and pressure on the security sources' endpoints.
+2.  It can be challenging to backport fixes and updates to previous releases.  This requires patch releases, which trigger the release workflow.  Improving the release process can be complex, as it may require patch release cuts, changes to different branches, and integration between logic that is managed in the master branch, executing other releases binaries.
+3.  The latest offline bundles does not support post `4.5` releases.  Bundle embedding for each release increases the cost to store and consume offline bundles.  The `roxctl` interface and documentation today needs to be clearer on these limitations.
+
+## Decision
+
+We will implement a versioning stream for vulnerability bundles, bumped when we determine that ClairCore's schema has changed.
+
+Initially, two version streams will exist (table 1 below):
+
+| Stream | Git Ref      | Description                                                        |
+|--------|--------------|--------------------------------------------------------------------|
+| dev    | heads/master | This stream tracks the latest vulnerability schema                 |
+| v1     | heads/4.4.5  | This stream tracks the schema used by the current StackRox release |
+
+On the subsequent releases (for example,. `4.4.6`), one of the two will be carried out:
+
+1.  The schema hasn't changed, so v1 now tracks the new release (`heads/4.4.6`).
+2.  The schema has changed, so v1 stays at the current release (`heads/4.4.5`), and a new version, v2, is created to track the next release (`heads/4.4.6`).
+
+### Building versioned bundles
+
+The GitHub workflow will have a map with "bundle version" as keys and "git reference" as values, similar to Table 1 above.  This can be stored statically in the matrix definition.
```

Comment:
so inside the workflow's YAML config?

Reply:
Is this manually updated?

Reply:
The specific location is not specified in the ADR. It could be in the YAML's config vs. a file in the repository.

> Is this manually updated?

Initially yes. It can be automated, AFAIK -- but I don't want to spread the decision here to other parts of the release automation workflows, scripts, etc.

Reply:
Any concerns here? Or just clarifying? Can we resolve?